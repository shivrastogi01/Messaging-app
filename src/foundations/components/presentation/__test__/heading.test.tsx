import React from "react";
import {
  Heading,
  HeadingVariants,
  HeadingSizes
} from "foundations/components/presentation";
import renderWithTheme from "main/testing/render-with-theme";

describe("Heading", () => {
  test("default variant matches the snapshot", () => {
    const { container } = renderWithTheme(<Heading>Heading</Heading>);
    expect(container.firstChild).toMatchSnapshot();
  });

  test("inverse variant matches the snapshot", () => {
    const { container } = renderWithTheme(
      <Heading variant={HeadingVariants.INVERSE}>Heading</Heading>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test("big size matches the snapshot", () => {
    const { container } = renderWithTheme(
      <Heading size={HeadingSizes.BIG}>Heading</Heading>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
