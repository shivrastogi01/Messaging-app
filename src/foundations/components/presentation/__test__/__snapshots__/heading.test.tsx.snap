// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading big size matches the snapshot 1`] = `
.c0 {
  color: inherit;
  font-family: inherit;
  font-size: 15px;
  font-weight: 500;
  font-size: 20px;
  font-weight: 700;
}

<h3
  class="c0"
>
  Heading
</h3>
`;

exports[`Heading default variant matches the snapshot 1`] = `
.c0 {
  color: inherit;
  font-family: inherit;
  font-size: 15px;
  font-weight: 500;
}

<h3
  class="c0"
>
  Heading
</h3>
`;

exports[`Heading inverse variant matches the snapshot 1`] = `
.c0 {
  color: inherit;
  font-family: inherit;
  font-size: 15px;
  font-weight: 500;
  color: #FFFFFF;
}

<h3
  class="c0"
>
  Heading
</h3>
`;
